overlord/snapstate/backend_test.go

```diff
@@ -931,6 +945,20 @@ func (f *fakeStore) Download(ctx context.Context, name, targetFn string, snapInf
 	return nil
 }
 
+func (f *fakeStore) DownloadIcon(ctx context.Context, name string, targetPath string, downloadURL string) error {
```

Comment:
this needs a pokeStateLock to make sure we are not keeping the state lock while downloading